MAINTAINING.md

```diff
@@ -0,0 +1,138 @@
+- [Updating the changelog](#updating-the-changelog)
+- [Documentation](#documentation)
+- [Versioning](#versioning)
+- [Releasing](#releasing)
+- [Labels](#labels)
+
+## Communication
+
+We have several ways that we can communicate with each other:
+
+- To show our direction and next steps, the [**roadmap**][roadmap] is the best place.
+- To track progress on the movement of issues, [**labels**](#labels)
+  are useful.
+  TODO: Fill with the Slack community and the regular community meeting
+
+[roadmap]: https://lyft.github.io/amundsen/roadmap/
+
+## Managing the community
+
+We try to create and foster a community around Amundsen. We do this by:
+
+- Answering questions from members of the community
+- Closing stale issues and feature requests
+- Keeping the community informed by ensuring that we add communications regularly with the new features
+- Ensuring that the documentation, as well as the documentation site, is kept up to
+  date
+
+## Workflow
+
+We generally follow [GitHub Flow]. The `master` branch is the main line, and all
+branches are cut from and get merged back into this branch. Generally, the
+workflow is as follows:
+
+[github flow]: https://help.github.com/articles/github-flow/
+
+- Cut a feature or bugfix branch from this branch.
+- Upon completing a branch, create a PR and ask another maintainer to approve
+  it.
+- Try to keep the commit history as clean as possible. Before merging, squash
+  "WIP" or related commits together and rebase as needed.
+- Once your PR is approved, and you've cleaned up your branch, you're free to
+  merge it in.
+
+## Architecture
+
+TODO
+
+## Running tests
+
+TODO
+
+## Updating the changelog
+
+**TO ADOPT?**
+
+After every user-facing change makes it into master, we make a note of it in the
+changelog, `CHANGELOG.md`. The changelog is sorted in reverse order by release version.
+
+Within each version, there are five available categories you can divide changes
+into. They are all optional:
+
+1. Deprecations
+1. Bug fixes
+1. Features
+1. Enhancements
+
+For each change, provide a human-readable description of the change.
+
+## Documentation
+
+We use [mkdocs] for creating our documentation from Markdown files. This system is configured
+from the 'mkdocs.yml' file in the root of this repository.
+
+Currently, our docs are built and deployed manually, so we should first build and deploy them locally and verify it. Here are some basic steps:
+
+1. Activate the virtualenv with `source venv/bin/activate`
+1. Install the requirements: `pip3 install -r requirements.txt`
+1. Install mkdocs with Homebrew (Mac only): `brew install mkdocs`
+1. Create the documentation with `mkdocs serve` navigate to them in localhost:8000. On mac OS, you may face ImportError and you may need to downgrade openssl by running `brew switch openssl 1.0.2r`
+1. Deploy our docs to the Github pages using `mkdocs gh-deploy`
+
+[mkdocs]: https://www.mkdocs.org/
+
+## Versioning
+
+### Naming a new version
+
+**TO REVIEW**
```

Comment:
Similarly here, this is want I _think_ we want to do but I don't think we've committed to it yet. If I remember correctly the conversation stalled around a couple concepts:
1. Whether or not we want to keep major versions in lock-step across the repositories (e.g. all 2.x.x versions of libraries will be compatible, all 3.x.x versions of libraries will be compatible).
2. If it would be too early to introduce this given tech debt we have across unifying the repos or needs of the community, which would force us (especially if combined with the item above) to run very quickly through major versions as we try to improve.

Will let others weigh in.

Reply:
yeap

Reply:
Yeah, I think some consistent versions across all repos would be really good. Otherwise, it'll become harder to make backwards-incompatible changes.
I think major versions should happen only for big changes. IMO, it's uncommon to go through more than 1 or 2 major versions in a year. If we think we'd be good through major versions very quickly perhaps we should do major and minor to be consistent across all repos and micros can be whichever way the sub-repo evolves.

Thoughts?